hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieClusteringJob.java

```diff
@@ -87,7 +87,7 @@ public static class Config implements Serializable {
     public int parallelism = 1;
     @Parameter(names = {"--spark-master", "-ms"}, description = "Spark master")
     public String sparkMaster = null;
-    @Parameter(names = {"--spark-memory", "-sm"}, description = "spark memory to use", required = true)
+    @Parameter(names = {"--spark-memory", "-sm"}, description = "spark memory to use", required = false)
     public String sparkMemory = null;
```

Comment:
Not very familiar with Spark, what is the behavior when default memory is not suffient for job execution then?

Reply:
The purpose of spark-memory is to set spark.executor.memory, which is usually configured according to  executor-memory in Spark. We shouldd configure both executor-memory and spark-memory if spark-memory is forced to true.

Reply:
you mean we lack option for `executor-memory` ? How the user can config that with current maste code?

Reply:
We use command 'spark-submit --executor-memory' can meet it, spark-memory is redundant.

Reply:
In the defallt way, we must use `spark-submit --executor-memory=4g --spark-memory=4g`.  However, we only use `spark-submit --executor-memory=4g` can meet it and `--spark-memory` should not keet it `required=true`.

Reply:
I'm wondering whether we should remove the option `--spark-memory` completely, like the usage of this options now is to override the memory value specified by `--executor-memory`, it has higher priority.

Reply:
I think remove the option `--spark-memory` is also a good choice. Setting executor memroy completely depends on `--executor-memory`.

Reply:
If coexistence of different values from `--executor-memory` and `--spark-memory` is allowed, then keeping the options sounds good to me, it gives some space for flexibility.